test/exercises/contribute_exercise_test.dart

```diff
@@ -53,41 +71,434 @@ void main() {
+
+  group('Form Field Validation Tests', () {
+    testWidgets('Exercise name field is required and displays validation error', (
+      WidgetTester tester,
+    ) async {
+      // Setup: Create verified user with required data
+      setupVerifiedUser();
+
+      // Build the exercise contribution screen
+      await tester.pumpWidget(createExerciseScreen());
+      await tester.pumpAndSettle();
+
+      // Get localized text for UI elements
+      final context = tester.element(find.byType(Stepper));
+      final l10n = AppLocalizations.of(context);
+
+      // Find the Next button (use .first since there are 6 steps with 6 Next buttons)
+      final nextButton = find.widgetWithText(ElevatedButton, l10n.next).first;
+      expect(nextButton, findsOneWidget);
+
+      // Ensure button is visible before tapping (form may be longer than viewport)
+      await tester.ensureVisible(nextButton);
+      await tester.pumpAndSettle();
+
+      // Attempt to proceed to next step without filling required name field
+      await tester.tap(nextButton);
+      await tester.pumpAndSettle();
+
+      // Verify that validation prevented navigation (still on step 0)
+      final stepper = tester.widget<Stepper>(find.byType(Stepper));
+      expect(stepper.currentStep, equals(0));
+    });
+
+    testWidgets('User can enter exercise name in text field', (WidgetTester tester) async {
+      // Setup: Create verified user
+      setupVerifiedUser();
+
+      // Build the exercise contribution screen
+      await tester.pumpWidget(createExerciseScreen());
+      await tester.pumpAndSettle();
+
+      // Find the first text field (exercise name field)
+      final nameField = find.byType(TextFormField).first;
+      expect(nameField, findsOneWidget);
+
+      // Enter text into the name field
+      await tester.enterText(nameField, 'Bench Press');
+      await tester.pumpAndSettle();
+
+      // Verify that the entered text is displayed
+      expect(find.text('Bench Press'), findsOneWidget);
+    });
+
+    testWidgets('Alternative names field accepts multiple lines of text', (
+      WidgetTester tester,
+    ) async {
+      // Setup: Create verified user
+      setupVerifiedUser();
+
+      // Build the exercise contribution screen
+      await tester.pumpWidget(createExerciseScreen());
+      await tester.pumpAndSettle();
+
+      // Find all text fields
+      final textFields = find.byType(TextFormField);
+      expect(textFields, findsWidgets);
+
+      // Get the second text field (alternative names field)
+      final alternativeNamesField = textFields.at(1);
+
+      // Enter multi-line text with newline character
+      await tester.enterText(alternativeNamesField, 'Chest Press\nFlat Bench Press');
+      await tester.pumpAndSettle();
+
+      // Verify that multi-line text was accepted and is displayed
+      expect(find.text('Chest Press\nFlat Bench Press'), findsOneWidget);
```

Comment:
we could also test here that the names are properly read into individual elements in the `alternateNamesEn` list